sentry-opentelemetry/sentry-opentelemetry-core/src/main/java/io/sentry/opentelemetry/SpanDescriptionExtractor.java

```diff
@@ -62,4 +66,117 @@ private OtelSpanInfo descriptionForDbSystem(final @NotNull ReadableSpan otelSpan
     @NotNull String description = dbStatement != null ? dbStatement : otelSpan.getName();
     return new OtelSpanInfo("db", description, TransactionNameSource.TASK);
   }
+
+  @SuppressWarnings("deprecation")
+  public @NotNull OtelSpanInfo extractSpanInfo(final @NotNull SpanData otelSpan) {
+    OtelSpanInfo spanInfo = extractSpanDescription(otelSpan);
+
+    final @Nullable Long threadId = otelSpan.getAttributes().get(SemanticAttributes.THREAD_ID);
+    if (threadId != null) {
+      spanInfo.addDataField("thread.id", threadId);
+    }
+
+    final @Nullable String threadName =
+        otelSpan.getAttributes().get(SemanticAttributes.THREAD_NAME);
+    if (threadName != null) {
+      spanInfo.addDataField("thread.name", threadName);
+    }
+
+    final @Nullable String dbSystem = otelSpan.getAttributes().get(SemanticAttributes.DB_SYSTEM);
+    if (dbSystem != null) {
+      spanInfo.addDataField("db.system", dbSystem);
+    }
+
+    final @Nullable String dbName = otelSpan.getAttributes().get(SemanticAttributes.DB_NAME);
+    if (dbName != null) {
+      spanInfo.addDataField("db.name", dbName);
+    }
+
+    return spanInfo;
+  }
+
+  @SuppressWarnings("deprecation")
+  private OtelSpanInfo extractSpanDescription(SpanData otelSpan) {
+    final @NotNull String name = otelSpan.getName();
+    final @NotNull Attributes attributes = otelSpan.getAttributes();
+
+    final @Nullable String httpMethod = attributes.get(SemanticAttributes.HTTP_METHOD);
+    if (httpMethod != null) {
+      return descriptionForHttpMethod(otelSpan, httpMethod);
+    }
+
+    final @Nullable String httpRequestMethod =
+        attributes.get(SemanticAttributes.HTTP_REQUEST_METHOD);
+    if (httpRequestMethod != null) {
+      return descriptionForHttpMethod(otelSpan, httpRequestMethod);
+    }
+
+    final @Nullable String dbSystem = attributes.get(SemanticAttributes.DB_SYSTEM);
+    if (dbSystem != null) {
+      return descriptionForDbSystem(otelSpan);
+    }
+
+    return new OtelSpanInfo(name, name, TransactionNameSource.CUSTOM);
+  }
+
+  @SuppressWarnings("deprecation")
+  private OtelSpanInfo descriptionForHttpMethod(
```

Comment:
Couldn't we have the overloaded method receiving `ReadableSpan` call this method with `ReadableSpan.toSpanData()`?